src/modules/deploy/provider/BaseProvider.ts

```diff
@@ -0,0 +1,15 @@
+export type Functions = Record<string, { url: string }>
+
+export abstract class BaseProvider {
+  public account: string
+
+  constructor(account: string) {
+    this.account = account
+  }
+
+  public abstract async createOrUpdateFunction(functionName: string, content: Buffer): Promise<void>
+
+  public abstract async createOrUpdateFunctions(functions: Record<string, Buffer>): Promise<void>
```

Comment:
Create types for the parameters like Function for the first and Function[] or Set<Function> for the second. Rename createOrUpdateFunctions according to the data structure like createOrUpdateFunctionList to have a more descriptive name, this also avoids confusion with createOrUpdateFunction.

Reply:
Good suggestion, function renamed!

It's harder to create a type to reuse on functions' parameters, though. An array will allow multiple functions with the same name. Similarly, a Set will allow multiple functions with same name if the objects have different references.

It needs to be an Object or Map, which will lead to redundancy: it could be `{ [functionName: string]: FunctionType }` where `type FunctionType = {functionName: string, content: Buffer}`, so we are repeating `functionName`.